src/main/java/uk/ac/cam/cl/dtg/isaac/dao/EventBookingPersistenceManager.java

```diff
@@ -111,8 +111,8 @@ public EventBookingDTO getBookingByEventIdAndUserId(final String eventId, final
      * @throws SegueDatabaseException
      *             - if an error occurs.
      */
-    public EventBookingDTO updateBookingStatus(final String eventId, final Long userId, final BookingStatus bookingStatus, final Map additionalEventInformation) throws SegueDatabaseException {
-        dao.updateStatus(eventId, userId, bookingStatus, additionalEventInformation);
+    public EventBookingDTO updateBookingStatus(final String eventId, final Long userId, final Long reservedById, final BookingStatus bookingStatus, final Map additionalEventInformation) throws SegueDatabaseException {
```

Comment:
What is the reservedById parameter for? I noticed null being passed in a few times in some method calls. If is nullable maybe we could annotated it as such / add explanation in the javadoc? I assume it is to let us record who made the booking/reservation? In which case maybe it could be called that and always populated - even if it is just the same as the userId?

Reply:
Makes sense unless there is business reason for having this information associated with a "reservation" event.

Reply:
There may be a business reason after all, in case we want to enforce the maximum limit of reservations per event, per teacher, on the number of participants that have been reserved by a certain teacher, rather than the ones who currently have a reservation that will be either confirmed or cancelled later. @mlt47 @jsharkey13